src/error.rs

```diff
@@ -2,19 +2,71 @@ use std::error::Error as StdError;
 use std::fmt;
 
 
-/// Describes possible errors originating from operations in this crate.
-pub type Error = GuardError;
+/// A transmutation error. This type describes possible errors originating
+/// from operations in this crate.
+///
+/// # Examples
+///
+/// ```
+/// # use safe_transmute::{ErrorReason, Error, guarded_transmute_bool_pedantic};
+/// # unsafe {
+/// assert_eq!(guarded_transmute_bool_pedantic(&[0x05]),
+///            Err(Error::InvalidValue));
+/// # }
+/// ```
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
+pub enum Error {
+    /// The data does not respect the target type's boundaries.
+    Guard(GuardError),
+    /// The given data slice is not properly aligned for the target type.
+    /// It would have been properly aligned if `offset` bytes were shifted
+    /// (discarded) from the front of the slice.
+    /// 
+    /// This is currently unused.
+    Unaligned {
+        offset: usize
+    },
+    /// The data contains an invalid value for the target type.
+    InvalidValue,
+}
+
+impl StdError for Error {
+    fn description(&self) -> &str {
+        match *self {
+            Error::Guard(ref e) => e.description(),
+            Error::Unaligned { .. } => "Unaligned data slice",
+            Error::InvalidValue => "Invalid target value",
+        }
+    }
+}
+
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match *self {
+            Error::Guard(ref e) => fmt::Display::fmt(e, f),
```

Comment:
Does `e.fmt(f)` not work or?

Reply:
Aye, it works as long as we don't include `Debug` in (which we don't have to because it's derived). It's in.